bears/c_languages/codeclone_detection/ClangComplexityBear.py

```diff
@@ -0,0 +1,88 @@
+from clang.cindex import Index, CursorKind
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result
+from bears.c_languages.ClangBear import clang_available
+
+
+class ClangComplexityBear(LocalBear):
+    """
+    Calculates cyclomatic complexity of each function and displays it to the
+    user.
+    """
+    check_prerequisites = classmethod(clang_available)
+    decisive_cursor_kinds = {
+        CursorKind.IF_STMT, CursorKind.WHILE_STMT, CursorKind.FOR_STMT,
+        CursorKind.DEFAULT_STMT, CursorKind.CASE_STMT}
+
+    def run(self, filename, file, max_complexity: int=10):
+        """
+        Calculates cyclomatic complexity of functions in file.
+
+        :param max_complexity:  Maximum cyclomatic complexity that is
+                                considered to be normal. The value of 10 had
+                                received substantial corroborating evidence.
+                                But the general recommendation: "For each
+                                module, either limit cyclomatic complexity to
+                                [the agreed-upon limit] or provide a written
+                                explanation of why the limit was exceeded."
+        """
+        self.complexities = {}
+        counter = {'decisions': 0, 'exits': 0}
```

Comment:
yes, technically, but from the perspective of the function it's global because it has state over all invocations of the recursive invocation which makes things complicated to think right?